### PR TITLE
[REEF-640]:Create CLI CPP project Org.Apache.REEF.ClrDriver. Phase I

### PR DESCRIPTION
**Pull request by beysims**
This addressed the issue by adding a managed cpp project to start clr driver

JIRA: [REEF-640](https://issues.apache.org/jira/browse/REEF-640)

This closes #

### CODE REVIEW

**Comment by markusweimer** [comment]
Why are there changes in the `.Client` project for this?

Also, can't this code be part of the `.Driver` project? Or does it have to be a new one because it is C++? If it is the latter: Maybe `DriverLauncher` or such is a better name? `ClrDriver` sounds as if this was somehow the CLR-based driver, which in fact is in `.Driver`.

**Comment by beysims** [comment]
Client project is modified in preparation for the next stage. 
DriverLauncher could be another name with the caveat that we would only be launching the driver on windows systems. 
We could try to use Driver project by making it an exe, but then it will not be possible to have multiple PRs, we would have to do it all in one PR.

**Comment by markusweimer** [comment]
Beysim and I just spoke about the naming thing and came up with `Org.Apache.REEF.Driver.Launcher` as the project and namespace name.

File: lang/cs/Org.Apache.REEF.ClrDriver/Org.Apache.REEF.ClrDriver.vcxproj [hunk 1]
```diff
@@ -0,0 +1,109 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="Build" ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.ClrDriver/Org.Apache.REEF.ClrDriver.vcxproj]: Adding license header

File: lang/cs/Org.Apache.REEF.ClrDriver/Org.Apache.REEF.ClrDriver.vcxproj [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>{0097E4AC-0CC9-450C-BC22-ACD5B2D55E70}</ProjectGuid>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <Keyword>ManagedCProj</Keyword>
+    <RootNamespace>OrgApacheREEFClrDriver</RootNamespace>
+  </PropertyGroup>
+  <Import Project="$(SolutionDir)\build.props" />
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'" Label="Configuration">
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.ClrDriver/Org.Apache.REEF.ClrDriver.vcxproj]: Do you still need this session as you have included build.prop or c++ project may have some thing special?

File: lang/cs/Org.Apache.REEF.ClrDriver/ReadMe.txt [hunk 3]
```diff
@@ -0,0 +1,35 @@
+========================================================================
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.ClrDriver/ReadMe.txt]: License header

File: lang/cs/Org.Apache.REEF.ClrDriver/Org.Apache.REEF.ClrDriver.cpp [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+}
+
+//
+// Invokes main method of entry class.
+// I.E. org/apache/reef/runtime/common/REEFLauncher
+//
+int CallMainMethodOfEntryClass(
+	JNIEnv* env,
+	char*	argv[],
+	int		firstArgOrdinal,
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.ClrDriver/Org.Apache.REEF.ClrDriver.cpp]: Remove tab space after int

File: lang/cs/Org.Apache.REEF.ClrDriver/Org.Apache.REEF.ClrDriver.cpp [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+	int		argCount)
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.ClrDriver/Org.Apache.REEF.ClrDriver.cpp]: same as above

File: lang/cs/Org.Apache.REEF.Client/Properties/Resources.xml [hunk 6]
```diff
@@ -85,6 +85,9 @@ This file is used to embed JAR files to client dll
   <data name="DriverJarFullName" xml:space="preserve">
     <value>reef-bridge-java-0.13.0-incubating-SNAPSHOT-shaded.jar</value>
   </data>
+  <data name="ClrDriverFullName" xml:space="preserve">
+    <value>Org.Apache.REEF.ClrDriver.exe</value>
+  </data>
   <assembly alias="System.Windows.Forms" name="System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" />
   <data name="reef_bridge_client" type="System.Resources.ResXFileRef, System.Windows.Forms">
     <value>E:\src\reef\lang\cs\bin\x64\Debug\Org.Apache.REEF.Bridge.JAR\reef-bridge-client-0.13.0-incubating-SNAPSHOT-shaded.jar;System.Byte[], mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Client/Properties/Resources.xml]: hard coded driver and folder?

File: lang/cs/Org.Apache.REEF.Client/Properties/Resources.xml [hunk 7]
```diff
@@ -92,4 +95,7 @@ This file is used to embed JAR files to client dll
   <data name="reef_bridge_driver" type="System.Resources.ResXFileRef, System.Windows.Forms">
     <value>E:\src\reef\lang\cs\bin\x64\Debug\Org.Apache.REEF.Bridge.JAR\reef-bridge-java-0.13.0-incubating-SNAPSHOT-shaded.jar;System.Byte[], mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Client/Properties/Resources.xml]: hard coded E:\...

File: lang/cs/Org.Apache.REEF.Client/Properties/Resources.xml [hunk 8]
```diff
@@ -92,4 +95,7 @@ This file is used to embed JAR files to client dll
...
   </data>
+  <data name="reef_clrdriver" type="System.Resources.ResXFileRef, System.Windows.Forms">
+    <value>E:\src\reef\lang\cs\bin\x64\Debug\Org.Apache.REEF.ClrDriver\Org.Apache.REEF.ClrDriver.exe;System.Byte[], mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Client/Properties/Resources.xml]: same as above

File: lang/cs/Org.Apache.REEF.Client/Org.Apache.REEF.Client.csproj [hunk 9]
```diff
@@ -109,6 +109,14 @@ under the License.
       <Project>{75503f90-7b82-4762-9997-94b5c68f15db}</Project>
       <Name>Org.Apache.REEF.Examples</Name>
     </ProjectReference>
+    <ProjectReference Include="..\Org.Apache.REEF.ClrDriver\Org.Apache.REEF.ClrDriver.vcxproj">
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Client/Org.Apache.REEF.Client.csproj]: use $(SolutionDir) to replace ..

**Comment by jwang98052** [comment]
I completed my pass.

File: lang/cs/Org.Apache.REEF.ClrDriver/Org.Apache.REEF.ClrDriver.vcxproj (hunk 2 above)
Comment (beysims) [inline lang/cs/Org.Apache.REEF.ClrDriver/Org.Apache.REEF.ClrDriver.vcxproj]: Not sure.

File: lang/cs/Org.Apache.REEF.ClrDriver/ReadMe.txt (hunk 3 above)
Comment (beysims) [inline lang/cs/Org.Apache.REEF.ClrDriver/ReadMe.txt]: Added

File: lang/cs/Org.Apache.REEF.ClrDriver/Org.Apache.REEF.ClrDriver.cpp (hunk 4 above)
Comment (beysims) [inline lang/cs/Org.Apache.REEF.ClrDriver/Org.Apache.REEF.ClrDriver.cpp]: fixed.

File: lang/cs/Org.Apache.REEF.ClrDriver/Org.Apache.REEF.ClrDriver.cpp (hunk 5 above)
Comment (beysims) [inline lang/cs/Org.Apache.REEF.ClrDriver/Org.Apache.REEF.ClrDriver.cpp]: fixed

File: lang/cs/Org.Apache.REEF.Client/Properties/Resources.xml (hunk 6 above)
Comment (beysims) [inline lang/cs/Org.Apache.REEF.Client/Properties/Resources.xml]: This gets changed during build time. Please see Org.Apache.REEF.Client.csproj

File: lang/cs/Org.Apache.REEF.Client/Properties/Resources.xml (hunk 7 above)
Comment (beysims) [inline lang/cs/Org.Apache.REEF.Client/Properties/Resources.xml]: his gets changed during build time. Please see Org.Apache.REEF.Client.csproj

File: lang/cs/Org.Apache.REEF.Client/Properties/Resources.xml (hunk 8 above)
Comment (beysims) [inline lang/cs/Org.Apache.REEF.Client/Properties/Resources.xml]: his gets changed during build time. Please see Org.Apache.REEF.Client.csproj

**Comment by beysims** [comment]
@jwang98052 I have completed this iteration

**Comment by jwang98052** [comment]
I will test and merge

**Pull request closed without merging**